Create stub template, change POLICIES.md/CONTRIBUTING.md (#1723)
exercism/java: POLICIES.md

> - Exercises of difficulty 4 or lower: provide stubs for all required constructors and methods. This means that you need to provide stubs for those constructors or methods that are necessary to pass all tests. E.g. stubs for private methods may be skipped.
Stubs should include the following body:
    `throw new UnsupportedOperationException("Delete this statement and write your own implementation.");`

> > - Exercises of difficulty 5 or higher: provide a starter file without stubs (StubTemplate.java file provided [here](https://github.com/exercism/java/tree/master/_template/src/main/java)) and rename it, and either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.

Reviewer: Sorry, I think this sentence is still a tiny bit confusing. Reading it I think it's not quite clear if you're meant to copy the provided file or create your own. Something like:
```
Exercises of difficulty 5 or higher: copy the StubTemplate.java file (provided [here](...)) and rename it, and either ...
```
might be more clear? 🙂 And maybe a sentence explaining what we'd like it to be renamed to could be useful? E.g. `... and rename it to fit the exercise. For example, for the exercise linked-list the file could be named LinkedList.java`. That's probably obvious, but maybe better to add too much information than too little? 🙂

Author: Yes, that's true, it's better to add more information. I added your suggestions.

> - Exercises of difficulty 4 or lower: provide stubs for all required constructors and methods. This means that you need to provide stubs for those constructors or methods that are necessary to pass all tests. E.g. stubs for private methods may be skipped.
Stubs should include the following body:
    `throw new UnsupportedOperationException("Delete this statement and write your own implementation.");`

> > - Exercises of difficulty 5 or higher: provide a starter file only (StubTemplate.java) and rename it, but either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.

Reviewer 2: ```suggestion
> - Exercises of difficulty 5 or higher: provide a starter file without stubs (StubTemplate.java) and rename it, and either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
```

> > - Exercises of difficulty 5 or higher: provide a starter file without stubs (StubTemplate.java) and rename it, and either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.

Reviewer: Might be useful to make it more clear where they should copy this file from? 🙂

Author: Thank you @FridaTveit , good point. I will add this information.
